dev-java/fop/fop-2.7.ebuild

```
@@ -0,0 +1,154 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+JAVA_PKG_IUSE="doc source test"
+MAVEN_ID="org.apache.xmlgraphics:fop:2.7"
+JAVA_TESTING_FRAMEWORKS="junit-4"
+
+inherit java-pkg-2 java-pkg-simple
+
+DESCRIPTION="XML Graphics Format Object Processor All-In-One"
+HOMEPAGE="https://xmlgraphics.apache.org/fop/"
+SRC_URI="mirror://apache/xmlgraphics/fop/source/fop-${PV}-src.tar.gz"
+
+LICENSE="Apache-2.0"
+SLOT="2.7"
+KEYWORDS="~amd64 ~ppc64 ~x86"
+
+CP_DEPEND="
+	dev-java/batik:1.14
+	dev-java/commons-io:1
+	dev-java/fontbox:0
+	dev-java/qdox:1.12
+	dev-java/sun-jai-bin:0
+	dev-java/xmlgraphics-commons:2
+"
+
+DEPEND="${CP_DEPEND}
+	>=virtual/jdk-1.8:*
+	dev-java/ant-core:0
+	dev-java/sun-jai-bin:0
+	test? (
+		dev-java/mockito:0
+		dev-java/pdfbox:0
+		dev-java/xmlunit:1
+	)
+"
+
+RDEPEND="${CP_DEPEND}
+	>=virtual/jre-1.8:*"
+
+DOCS=( NOTICE README )
+
+S="${WORKDIR}/fop-${PV}"
+
+src_prepare() {
+	default
+	java-pkg_clean
+}
+
+src_compile() {
+	JAVA_SRC_DIR="fop-util/src/main/java"
+	JAVA_JAR_FILENAME="fop-util.jar"
+	java-pkg-simple_src_compile
+	JAVA_GENTOO_CLASSPATH_EXTRA+=":fop-util.jar"
+	rm -fr target || die
+
+	JAVA_SRC_DIR="fop-events/src/main/java"
+	JAVA_RESOURCE_DIRS="fop-events/src/main/resources"
+	JAVA_JAR_FILENAME="fop-events.jar"
+	java-pkg-simple_src_compile
+	JAVA_GENTOO_CLASSPATH_EXTRA+=":fop-events.jar"
+	rm -fr target || die
+
+	JAVA_SRC_DIR="fop-core/src/main/java"
+	JAVA_RESOURCE_DIRS="fop-core/src/main/resources"
+	JAVA_MAIN_CLASS=$( sed -n 's:.*<mainClass>\(.*\)</mainClass>:\1:p' fop-core/pom.xml )
+	JAVA_JAR_FILENAME="fop-core.jar"
+	# Code generation according to
+	# https://github.com/apache/xmlgraphics-fop/blob/fop-2_7/fop-core/pom.xml#L150-L219
+	pushd fop-core/src/main/codegen/fonts
+		local fonts=$(find . -name "Courier*.xml" \
+			-o -name "Helvetica*.xml" \
+			-o -name "Times*.xml" \
+			-o -name "Symbol.xml" \
+			-o -name "ZapfDingbats.xml"
+			)
+		for font in ${fonts}; do \
+			xalan -XSLTC \
+				-IN $font \
+				-XSL font-file.xsl \
+				-OUT ../../java/org/apache/fop/fonts/base14/${font//.xml}.java || die
+		done
+		xalan -XSLTC \
+			-IN encodings.xml \
+			-XSL code-point-mapping.xsl \
+			-OUT ../../java/org/apache/fop/fonts/base14/CodePointMapping.java || die
+	popd
+	java-pkg-simple_src_compile
+	rm -fr target || die
```

That `|| die` is strange here, as IIRC `rm -f` never fails. Assuming that `target` is a directory, I wonder if
```suggestion
	rm -r target || die
```
wouldn't simply work?

---

Good catch. Never really tried it, just blindly copied from `tomcat-servlet-api` which I used as a template since I first time tried a multi-jar ebuild for `jsch-agent-proxy`.
All three occurancies adjusted.